### ISSUE TITLE
[v4] | [v2] application build crashing on real device

### ISSUE THREAD

**Issue by ziishaned** [gorhom/react-native-scroll-bottom-sheet#1526]
# Bug

The application crashes on the real device upon opening bottom sheet. Here is the error which I am getting:

```
RangeError: Maximum call stack size exceeded (native stack depth)
    at anonymous (JavaScript:1:288)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    ... skipping 157 frames
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at apply (native)
    at anonymous (JavaScript:1:336)
    at anonymous (JavaScript:1:200)
    at anonymous (JavaScript:1:95)
    at mapperRun (JavaScript:1:1040)
    at anonymous (JavaScript:1:350)
    at callMicrotasksOnUIThread (JavaScript:1:61)
    at anonymous (JavaScript:1:139)
    at com.swmansion.reanimated.AndroidUIScheduler.triggerUI(Native Method)
    at com.swmansion.reanimated.AndroidUIScheduler$1.run(AndroidUIScheduler.java:24)
    at com.swmansion.reanimated.AndroidUIScheduler$2.runGuarded(AndroidUIScheduler.java:43)
    at com.facebook.react.bridge.GuardedRunnable.run(GuardedRunnable.java:30)
    at android.os.Handler.handleCallback(Handler.java:958)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loopOnce(Looper.java:205)
    at android.os.Looper.loop(Looper.java:294)
    at android.app.ActivityThread.main(ActivityThread.java:8177)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
```

## Environment info

| Library                         | Version |
| ------------------------------- | ------- |
| @gorhom/bottom-sheet            | 4.5.0   |
| react-native                    | 0.72.4   |
| react-native-reanimated         | 3.5.1   |
| react-native-gesture-handler    | 2.12.1   |

## Steps To Reproduce

1. Create the application build
2. Run the build on the real device 
3. Open the app and navigate to the page where the bottom sheet code is
4. Try to open the bottom sheet.

Describe what you expected to happen:

The application should not crash and the bottom sheet should open normally.

## Reproducible sample code

```tsx
import { View, Text, Button } from "react-native";
import { BottomSheetBackdrop, BottomSheetModal } from "@gorhom/bottom-sheet";

function App() {
  const bottomSheetModalRef = useRef<BottomSheetModal>(null);
  const [bottomSheetSnapIndex, setBottomSheetSnapIndex] = useState<number>();

  const renderBackdrop = (backdropProps: BottomSheetBackdropProps) => (
    <BottomSheetBackdrop
      {...backdropProps}
      appearsOnIndex={0}
      disappearsOnIndex={-1}
    />
  );

  return (
    <>
      <Button
        color="black"
        title="Present Modal"
        onPress={() => bottomSheetModalRef.current?.present()}
      />
      <BottomSheetModal
        index={0}
        handleComponent={null}
        ref={bottomSheetModalRef}
        snapPoints={["50%", "90%"]}
        enablePanDownToClose={false}
        enableHandlePanningGesture={false}
        backdropComponent={renderBackdrop}
        enableContentPanningGesture={false}
        onChange={(index) => setBottomSheetSnapIndex(index)}
      >
        <View>
          <Text>Awesome 🎉</Text>
        </View>
      </BottomSheetModal>
    </>
  );
}

```

**Comment by jbagaresgaray** [comment]
The issue above happens on Android whereas this is crash issue for iOS

```
C++ Exception: N8facebook3jsi7JSErrorE
```

<h2 dir="auto" style="box-sizing: border-box; margin-top: 24px; margin-bottom: 16px; font-size: 1.5em; font-weight: var(--base-text-weight-semibold, 600); line-height: 1.25; padding-bottom: 0.3em; border-bottom: 1px solid var(--borderColor-muted, var(--color-border-muted)); color: rgb(31, 35, 40); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Environment info</h2>

Library | Version
-- | --
@gorhom/bottom-sheet | 4.5.0
react-native | 0.71.13
react-native-reanimated | 3.5.1
react-native-gesture-handler | 2.12.1

<br class="Apple-interchange-newline">

**Comment by cvthien-dev** [comment]
same issue, downgrade react-native-reanimated to 3.4.2 work for me

**Comment by puniker** [comment]
> same issue, downgrade react-native-reanimated to 3.4.2 work for me

its worked for me. Thanks!

**Comment by ziishaned** [comment]
Closing this issue as it is fixed in the latest release `3.5.2` of `react-native-reanimated` package https://github.com/software-mansion/react-native-reanimated/pull/5087
